src/index.ts

```diff
@@ -0,0 +1,330 @@
+       */
+      this.loadNftData(tokenData);
+
+    } catch (error: any) {
+      this.api.notifier.show({
+        message: error.message,
+        style: 'error',
+      });
+    }
+  }
+
+  private fetchNft() {
+    const network = this.nodes.formNetworkSelect.value;
+    const contractAddress = this.nodes.formContractAddressInput.value;
+    const tokenId = this.nodes.formTokenIdInput.value;
+
+    const tokenData = {
+      network,
+      contractAddress,
+      tokenId,
+    };
+
+    this.loadNftData(tokenData);
+  }
+
+  private async loadNftData(tokenData: {network: string, contractAddress: string, tokenId: string}): void {
+    const { network, contractAddress, tokenId } = tokenData;
+
+    if (!this.config.endpoint) {
+      this.api.notifier.show({
+        message: 'Endpoint API is not set',
+        style: 'error',
+      });
+      return;
+    }
+
+    try {
+      const requestData: NftToolServerRequest = {
+        network,
+        contractAddress,
+        tokenId,
+      }
+
+      const response = await fetch(this.config.endpoint, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(requestData),
+      });
+
+      const data = await response.json() as NftToolServerResponse;
+
+      if (!data.success) {
+        throw new Error(data.message);
+      }
+
+      this.data = data.message as NftToolServerResponseData;
+      this.fillNftCard();
+
+    } catch (error: any) {
+      this.api.notifier.show({
+        message: error.message,
+        style: 'error',
+      });
+    }
+  }
+
+  /**
+   * Fill NFT card with data
+   */
+  private fillNftCard(): void {
+    /**
+     * If media is a video, use video tag
+     */
+    if (this.data.media.endsWith('mp4')) {
+      this.nodes.imageElement = Dom.make('video', styles['media'], {
+        src: this.data.media,
+        playsinline: true,
+        muted: true,
+        autoplay: true,
+        loop: true,
+      });
+    } else {
+      this.nodes.imageElement = Dom.make('img', styles['media'], {
+        src: this.data.media,
+      });
+    }
+
+    if (this.nodes.media) {
+      this.nodes.media.innerHTML = '';
+      this.nodes.media.appendChild(this.nodes.imageElement);
+    }
+
+    if (this.nodes.collection) this.nodes.collection.innerText = this.data.collection;
+    if (this.nodes.title) this.nodes.title.innerText = this.data.title;
+
+    if (this.nodes.formNetworkSelect) {
+      this.nodes.formNetworkSelect.value = this.data.network;
+    }
+
+    if (this.nodes.formContractAddressInput) {
+      this.nodes.formContractAddressInput.value = this.data.contractAddress;
+    }
+
+    if (this.nodes.formTokenIdInput) {
+      this.nodes.formTokenIdInput.value = this.data.tokenId;
+    }
+  }
+
+  /**
+   * Regexp for a opensea link
+   */
+  private static get regexp(): RegExp {
+    return /^https:\/\/opensea\.io\/assets\/([a-zA-Z]+)\/([A-Za-z0-9]+)\/(([A-Za-z0-9]+))$/i;
+  }
+};
```

Comment:
Please, add .editorconfig file. Because there is no empty final line in your files.